### DISCUSSION TITLE
Query on Variational Graph Auto-Encoder

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#4341] [Q&A] [answered]
Hi,
My question is, how can I split the latent representation of GVAE into two parts? One is for nodes, and the other is for edges. Is this splitting of the latent representation of GVAE possible? I am waiting for your positive reply.

**Comment by rusty1s** [comment, marked as answer]
Can you clarify what you want to try to do? Note that `VGAE` only computes latent representations for nodes, and uses them to reconstruct the adjacency matrix. It should be nonetheless possible to extend upon this scheme to compute both node and edge representations using a GNN, and using both to reconstruct the input.

**Reply by shrimonmuke0202** [reply]
Thanks for your Answer,
How can I get node and edge representation from latent representation using a GNN as you said that VGAE only computes latent representations for nodes?

**Reply by rusty1s** [reply]
Do you mean you want to reconstruct node and edge features? This should be totally possible. You can just use a linear layer/MLP to reconstruct node features, and you can do the same for pairwise node embeddings to reconstruct edge features. You can then combine all three objectives (node features, edge features and link existence) and train them jointly.

**Reply by shrimonmuke0202** [reply]
I want to reconstruct node and edge features from the latent representation. Thus, I have to use a linear layer on the latent representation to get node and edge embedding. Also, can you give me an example of how I can apply GVAE to the ZINC dataset i.e., in the molecular graph generation task?

**Reply by rusty1s** [reply]
I sadly do not have an implementation of this, but happy to help out if you have any questions on how to do so.

**Reply by shrimonmuke0202** [reply]
In [this](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/autoencoder.py) example, VGAE is applied on a single graph. My query is how to process the same thing on batched graphs (multiple graphs in a single batch)?

**Reply by rusty1s** [reply]
How about
```python
dataset = MyDataset(..., transform=RandomLinkSplit(...))
train_data_list = []
for train_data, val_data, test_data in dataset:
    train_data_list.append(train_data)

train_loader = DataLoader(train_data_list, batch_size=..., shuffle=True)
```
and then apply the pipelines as given (while iterating over all data inside `train()`). Importantly, you may need to provide your own negative samples (via `batched_negative_sampling`) [here](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/autoencoder.py#L81).

**Reply by shrimonmuke0202** [reply]
Thanks for your comments; this will be very helpful for me. If there is any problem during the implementation of the code, then I will ping in the following thread.
